src/xwayland.c

```diff
@@ -979,6 +991,25 @@ handle_xwm_ready(struct wl_listener *listener, void *data)
 	xwayland_update_workarea(server);
 }
 
+void
+xwayland_create_window_type_rules(void)
+{
+	// Desktop window type
+	struct window_rule *rule = znew(*rule);
+	rule->window_type = NET_WM_WINDOW_TYPE_DESKTOP;
+	rule->server_decoration = LAB_PROP_FALSE;
+	rule->skip_taskbar = LAB_PROP_TRUE;
+	rule->skip_window_switcher = LAB_PROP_TRUE;
+	rule->ignore_focus_request = LAB_PROP_TRUE;
```

Comment:
I'm not so sure either, but as far as I can understand ignore_focus_request setting makes it so that window unable to take over user focus? If that's the case, can you give an example in which desktop window might want to take over user focus? In my opinion it's not very convenient for users to stumble upon desktop which randomly focuses itself

Reply:
I haven't tested, but imagine an desktop client showing icons (like `pcmanfm --desktop` and `xfdesktop`) would want to take focus to operate right-click menus, rename icons and so on.

Reply:
Yeah, it needs focus to do that of course, but it's user-initiated focus, rather than focus requested by window, isn't it? Sorry if I'm talking nonsense, that's just what I figured from the setting name without looking deep into implementation. This is also confirmed by the fact that I can navigate right-click menus and rename files with this setting set to true.